yellowbrick/text/postag.py

```diff
@@ -114,251 +115,381 @@ def __init__(
+                            self.pos_tag_counts_["noun"] += 1
+                        elif tag.startswith("J"):
+                            self.pos_tag_counts_["adjective"] += 1
+                        elif tag.startswith("V"):
+                            self.pos_tag_counts_["verb"] += 1
+                        # include particles with adverbs
+                        elif tag.startswith("RB") or tag == "RP":
+                            self.pos_tag_counts_["adverb"] += 1
+                        elif tag.startswith("PR"):
+                            self.pos_tag_counts_["pronoun"] += 1
+                        elif tag.startswith("W"):
+                            self.pos_tag_counts_["wh- word"] += 1
+                        elif tag == "CC":
+                            self.pos_tag_counts_["conjunction"] += 1
+                        elif tag == "CD":
+                            self.pos_tag_counts_["digit"] += 1
+                        # combine predeterminer and determiner
+                        elif tag in ["DT" or "PDT"]:
+                            self.pos_tag_counts_["determiner"] += 1
+                        elif tag == "EX":
+                            self.pos_tag_counts_["existential"] += 1
+                        elif tag == "FW":
+                            self.pos_tag_counts_["non-English"] += 1
+                        elif tag == "IN":
+                            self.pos_tag_counts_["preposition"] += 1
+                        elif tag == "POS":
+                            self.pos_tag_counts_["possessive"] += 1
+                        elif tag == "LS":
+                            self.pos_tag_counts_["list"] += 1
+                        elif tag == "MD":
+                            self.pos_tag_counts_["modal"] += 1
+                        elif tag in self.punct_tags:
+                            self.pos_tag_counts_["punctuation"] += 1
+                        elif tag == "TO":
+                            self.pos_tag_counts_["infinitive"] += 1
+                        elif tag == "UH":
+                            self.pos_tag_counts_["interjection"] += 1
+                        elif tag == "SYM":
+                            self.pos_tag_counts_["symbol"] += 1
+                        else:
+                            self.pos_tag_counts_["other"] += 1
 
     def draw(self, **kwargs):
         """
         Called from the fit method, this method creates the canvas and
         draws the part-of-speech tag mapping as a bar chart.
-
         Parameters
         ----------
         kwargs: dict
             generic keyword arguments.
-
         Returns
         -------
         ax : matplotlib axes
             Axes on which the PosTagVisualizer was drawn.
         """
-        colors = resolve_colors(
-            n_colors=len(self.pos_tag_counts_),
-            colormap=self.colormap,
-            colors=self.colors,
-        )
-
-        if self.frequency:
+        if self.stack:
```

Comment:
In this case, the control logic is probably at an appropriate place, but there is a lot of duplicated code here as well. 

There seem to be four cases to consider:

1. stack/frequency
2. stack/no-frequency
3. no-stack/frequency
4. no-stack/no-frequency 

What code do each of these cases share? What are their primary differences? Do you think you could take another look to see if you can specialize each case (and comment that they exist and which case is which) or to reduce the duplicate code? I'm happy to discuss it further if you'd like - I know I have a number of tricks up my sleeve for handling this and would be happy to chat about it!